src/index.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022-present Mirage Aegis
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { readdirSync } from "node:fs";
+import { join } from "node:path";
+import { Client, Collection, Events, GatewayIntentBits, Interaction } from "discord.js";
+import { Command } from "./util/command-template.js";
+
+// Loads the environment variables
+require("dotenv").config();
+
+/**
+ * Wrapper class adding a collection of slash commands
+ * to a Discord Client
+ */
+class Bot extends Client {
+    /**
+     * A collection of slash commands that the bot has
+     */
+    public commands: Collection<string, Command>;
+}
+
+// Client instance with all required intents
+const client = new Bot({
+    intents: [
+        GatewayIntentBits.Guilds,
+        GatewayIntentBits.MessageContent,
+        GatewayIntentBits.GuildMembers,
+        GatewayIntentBits.GuildMessages,
+    ],
+    presence: {
+        status: "online",
+        activities: [{
+            name: "The World of sushis!",
+            type: 0  // Playing
+        }]
+    }
+});
```

Comment:
The bot starts here, with the required intents (see [gateway intents](https://discord.com/developers/docs/topics/gateway#gateway-intents)) and the status of the bot